R/pcens.R

```diff
@@ -13,13 +13,13 @@
 #' @export
 new_pcens <- function(
     pdist, dprimary, dprimary_args,
-    pdist_name = NULL,
-    dprimary_name = NULL, ...) {
-  if (is.null(pdist_name)) {
-    pdist_name <- .extract_function_name(substitute(pdist))
+    ...) {
+
+  if (is.null(attr(pdist, "name"))) {
+    attr(pdist, "name") <- .extract_function_name(pdist)
   }
```

Comment:
a possibly desirable alternative: push the conditional check into .extract_function_name - since it no longer depends on NSE name parsing, could make this more like `pdist <- .ensure_function_name(pdist)` and push the check / attribute setting into the utility function.

that would encapsulate the use of attributes, and concentrate future handling of other, non-`stats` distribution sources (somewhat - also need to deal w/ how the attr is *used* elsewhere in the code)

Reply:
could also make it be side-effect-y / pass-by-reference style, and have `.ensure_function_name(...)` modify items in parent environment, if there are any performance issues here with having to copy-modify-reassign or if that sort of approach is otherwise preferrable stylistically / maintenance wise.